OsmAnd/src/net/osmand/plus/search/dialogs/QuickSearchDialogFragment.java

```diff
@@ -973,6 +973,7 @@ public void onDismiss(@NonNull DialogInterface dialog) {
 				fragmentManager.popBackStack();
 			}
 		}
+		app.getOsmandMap().getMapLayers().getNearbyPlacesLayer().customObjectsDelegate.setCustomMapObjects(null);
```

Comment:
Is this place correct, won`t screen rotation break it? onBackButtonPressed()?

also we could try to use similar logic in NearbyPlacesHelper as in PoiFiltersHelper, without using directly NearbyPlacesLayer